### ISSUE TITLE
stylefunction.js' fromTemplate might receive an array

### ISSUE THREAD

**Issue by aberenyi** [boundlessgeo/ol-mapbox-style#87]
Hiya,

Just came across this issue where the `stylefunction.js`' [`fromTemplate`](https://github.com/boundlessgeo/ol-mapbox-style/blob/master/stylefunction.js#L139) function receives an `array` as it's `text` argument, therefore it throws a wobbly. 

```
Uncaught TypeError: text.match is not a function
```

Within Mapbox Studio one might use a function (`to-string()`) to convert non-string fields to string fields and use those as labels, hence the issue.

The text argument looks something like this:
```
["to-string", ["get", "<attributeName>"]]
```

A quick and dirty workaround can look like this:
```js
if (Array.isArray(text) && text.length === 2 && text[0] === 'to-string' && Array.isArray(text[1]) && text[1].length === 2 && text[1][0] === 'get') {
  text = properties[text[1][1]].toString();
}
```

**Comment by aberenyi** [comment]
It looks like the returned value is an array even if the label is defined as a single string field (i.e. without a function like `to-string`).
```
["get", "<attributeName>"]
```
Moreover, there are a lot of other functions available, like `length`, `match`, `any`, etc.

**Comment by ahocevar** [comment]
Have you tried this with latest master? #86 added support for these expressions, so your styles should work now.

**Comment by aberenyi** [comment]
That's great news.

Sadly I could not try the latest master as both `npm i ol-mapbox-style@latest` and `npm i` in a freshly cloned repo fails with
```
./util/has_lib.sh: line 31: pkg-config: command not found
```
on macOS.

**Comment by aberenyi** [comment]
Sorry, I meant `npm i boundlessgeo/ol-mapbox-style` not `npm i ol-mapbox-style@latest`.

**Comment by ahocevar** [comment]
That's the `canvas` dev dependency. Try `npm install --production ol-mapbox-style@beta`.

**Comment by aberenyi** [comment]
After installing `pkg-config` and `cairo` w/ Homebrew both install methods work just fine.

However, I'm back to square one, as I get the same error even with `ol-mapbox-style@3.0.0-beta.8`.

**Comment by ahocevar** [comment]
Can you share a validated style document that triggers the error?

**Comment by aberenyi** [comment]
Sure thing - [style.json.zip](https://github.com/boundlessgeo/ol-mapbox-style/files/2102660/style.json.zip).

Validates fine against `mapbox-gl-style-spec` v12.0.0.

**Comment by ahocevar** [comment]
Thanks @aberenyi. I just pushed d605f0dd9d2827735f7bdd6ff47a7c2824f81828 and released v3.0.0-beta.9. The problem was that we tried to convert expressions to expressions.

Can you please verify that things work now and report back?

**Comment by aberenyi** [comment]
Works like a charm - many thanks @ahocevar!
